clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/caching/agents/TitusStreamingUpdateAgent.java

```diff
@@ -322,6 +328,18 @@ private void updateTask(StreamingCacheState state, Task task) {
       state.changes.incrementAndGet();
     }
 
+    private void updateMovedTask(StreamingCacheState state, Task task, String destinationJobId,
+                                 String sourceJobId) {
+      if (sourceJobId != null) {
+        if (state.tasks.containsKey(sourceJobId)) {
+          state.tasks.get(destinationJobId).add(task);
+          state.tasks.get(sourceJobId).remove(task);
```

Comment:
Check that `source.tasks.containsKey(sourceJobId)` before removing. This doesn't look like it's necessary, but there isn't a guarantee of event ordering or only-once delivery. Note the "..but task not present in current snapshot set" debug logging in `updateTask()` - we see this fire in prod occasionally as a result.

Reply:
updated per your inputs